projects/baby_driver/scripts/can_util.py

```diff
@@ -183,3 +183,33 @@ def next_message(
         ))
 
     return msg
+
+def can_pack(data_list):
+    """
+    Converts list of tuples and combines them into a little-endian
+    bytearray rendition. List is traversed in reverse order, each val
+    is broken into bytes and appended to bytearr output (LSB first)
+    Args:
+        List of tuples in form ((int) val, (int) len_in_bytes). val must be
+        in range [0, 2**len_in_bytes - 1]
+    Returns:
+        A bytearray object in little endian format, representing
+        message components input
+    Raises:
+        ValueError: if invalid values for val, len_in_bytes input
+    """
+    bytearr = bytearray()
+    # Traverse list in reverse order
+    for val, len_in_bytes in reversed(data_list):
```

Comment:
Wait, perhaps I misexplained. By "little-endian order" I meant that each val should be encoded in little-endian order but the vals should still be in the order given in data_list. So `can_pack([(0xABCDEF, 4), (0x010203, 3)])` should be equal to `bytearray([0xEF, 0xCD, 0xAB, 0x00, 0x03, 0x02, 0x01])`. So there's no need to traverse the list in reverse order. Sorry for the confusion.